src/Compilers/VisualBasic/Portable/Binding/Binder_Expressions.vb

```diff
@@ -303,6 +301,147 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End If
         End Function
 
+        Private Function BindTupleExpression(node As TupleExpressionSyntax, diagnostics As DiagnosticBag) As BoundExpression
+            Dim arguments As SeparatedSyntaxList(Of SimpleArgumentSyntax) = node.Arguments
+            Dim numElements As Integer = arguments.Count
+
+            If numElements < 2 Then
+                ' this should be a parse error already.
+                Dim args = If(numElements = 1,
+                    ImmutableArray.Create(Of BoundNode)(BindRValue(arguments(0).Expression, diagnostics)),
+                    ImmutableArray(Of BoundNode).Empty)
+
+                Return BadExpression(node, args, ErrorTypeSymbol.UnknownResultType)
+            End If
+
+            Dim hasErrors = False
+            Dim hasNaturalType = True
+
+            ' set of names already used
+            Dim uniqueFieldNames = PooledHashSet(Of String).GetInstance()
+
+            Dim boundArguments = ArrayBuilder(Of BoundExpression).GetInstance(arguments.Count)
+            Dim elementTypes = ArrayBuilder(Of TypeSymbol).GetInstance(arguments.Count)
+            Dim elementLocations = ArrayBuilder(Of Location).GetInstance(arguments.Count)
+
+            Dim elementNames As ArrayBuilder(Of String) = Nothing
+            Dim countOfExplicitNames = 0
+
+            ' prepare And check element names And types
```

Comment:
fixed

---
In reply to: [71439851](https://github.com/dotnet/roslyn/pull/12596#discussion_r71439851) [](ancestors = 71439851)